diff --git a/dist/machine/oid4vciMachine.js b/dist/machine/oid4vciMachine.js
index 7f4e2e4b94e3647d4029bff4b95294e877ee0d14..c52579ffbd390b9c5e1605184421795b3c9d1bc6 100644
--- a/dist/machine/oid4vciMachine.js
+++ b/dist/machine/oid4vciMachine.js
@@ -60,6 +60,11 @@ const oid4vciRequireAuthorizationGuard = (ctx, _event) => {
     if (!openID4VCIClientState.authorizationURL) {
         return false;
     }
+    else if (openID4VCIClientState.credentialOffer &&
+      openID4VCIClientState.credentialOffer.preAuthorizedCode &&
+      ((_a = openID4VCIClientState.credentialOffer.supportedFlows) === null || _a === void 0 ? void 0 : _a.includes(oid4vci_common_1.AuthzFlowType.PRE_AUTHORIZED_CODE_FLOW))) {
+        return false;
+    }
     else if (openID4VCIClientState.authorizationRequestOpts) {
         // We have authz options or there is not credential offer to begin with.
         // We require authz as long as we do not have the authz code response
diff --git a/src/machine/oid4vciMachine.ts b/src/machine/oid4vciMachine.ts
index 7862ff13aa798973bdcd4b7128473f6aab057305..ece65effc0435bb549e29e69561bc52efb612df3 100644
--- a/src/machine/oid4vciMachine.ts
+++ b/src/machine/oid4vciMachine.ts
@@ -84,6 +84,12 @@ const oid4vciRequireAuthorizationGuard = (ctx: OID4VCIMachineContext, _event: OI
 
   if (!openID4VCIClientState.authorizationURL) {
     return false
+  } else if (
+    openID4VCIClientState.credentialOffer &&
+    openID4VCIClientState.credentialOffer.preAuthorizedCode &&
+    openID4VCIClientState.credentialOffer.supportedFlows?.includes(AuthzFlowType.PRE_AUTHORIZED_CODE_FLOW)
+  ) {
+    return false
   } else if (openID4VCIClientState.authorizationRequestOpts) {
     // We have authz options or there is not credential offer to begin with.
     // We require authz as long as we do not have the authz code response
