diff --git a/browser/driver/DriverFactory.js b/browser/driver/DriverFactory.js
index fa19d4e4605f0d0346afb540f0bb7efb52e8b3b2..6b78fad879066284d5b83d6a18bb0938fa59a174 100644
--- a/browser/driver/DriverFactory.js
+++ b/browser/driver/DriverFactory.js
@@ -5,7 +5,6 @@ import { SqlServerDriver } from "./sqlserver/SqlServerDriver";
 import { OracleDriver } from "./oracle/OracleDriver";
 import { SqliteDriver } from "./sqlite/SqliteDriver";
 import { CordovaDriver } from "./cordova/CordovaDriver";
-import { ReactNativeDriver } from "./react-native/ReactNativeDriver";
 import { NativescriptDriver } from "./nativescript/NativescriptDriver";
 import { SqljsDriver } from "./sqljs/SqljsDriver";
 import { MysqlDriver } from "./mysql/MysqlDriver";
@@ -45,8 +44,6 @@ export class DriverFactory {
                 return new CordovaDriver(connection);
             case "nativescript":
                 return new NativescriptDriver(connection);
-            case "react-native":
-                return new ReactNativeDriver(connection);
             case "sqljs":
                 return new SqljsDriver(connection);
             case "oracle":
